scripts/format_code.py

```diff
@@ -0,0 +1,257 @@
+  """Executes clang-format on the file to reformat it according to our
+  project's standards.  If -format_in_place is defined, the file will be
+  formatted in place, with the results overwriting the previous version
+  of the file. Otherwise the result of the file reformat will be logged
+  to standard out.
+
+  Args:
+   filename: path to the file to format.
+  """
+  command_line = CLANG_FORMAT_BASE + CLANG_FORMAT_PARM_IN_PLACE + filename
+  args = command_line.split(" ")
+  proc = subprocess.Popen(args)
+
+def git_diff_list_files():
+  """Compares the current branch to master to assemble a list of source
+  files which have been altered.
+  
+  Returns:
+    A list of file paths for each file in the git diff list with an extension
+    matching one of those in FILE_TYPE_EXTENSIONS.
+  """
+  filenames = []
+  command_line = (GIT_DIFF_BASE + FLAGS.git_range + GIT_DIFF_PARAMS).strip()
+  args = command_line.split(" ")
+  if FLAGS.verbose:
+    print()
+    print("Executing git diff to detect changed files, git_range:",
+      "\"" + FLAGS.git_range + "\"") 
+  proc = subprocess.Popen(args, stdout=subprocess.PIPE)
+  for line in io.TextIOWrapper(proc.stdout, encoding="utf-8"):
+    line = line.rstrip()
+    if(line.endswith(FILE_TYPE_EXTENSIONS)):
+      filenames.append(line.strip())
+      if FLAGS.verbose:
+        print("  - File detected: " + line)
+  if filenames and FLAGS.verbose:
+    print()
+  return filenames
+
+def list_files_from_directory(path, recurse):
+  """Iterates through the the directory and returns a list of files
+  which match those with extensions defined in FILE_TYPE_EXTENSIONS. 
+
+  Args:
+    path: the path to a directory to start searching from.
+    recurse: when True, will also recursively search for files in any
+      subdirectories found in path.
+  
+  Returns:
+    A list of matching filenames found in the directory.
+  """
+  filenames = []
+  for root, dirs, files in os.walk(path):
+    path = root.split(os.sep)
+    for filename in files:
+      if filename.endswith(FILE_TYPE_EXTENSIONS):
+        full_path = os.path.join(root, filename)
+        if FLAGS.verbose:
+          print("\t\t" + full_path)
+        filenames.append(full_path)  
+      if not recurse:
+        break;
+  return filenames
+
+def directory_search_list_files():
+  """Create a list of files based on the directories defined in the command
+  line arguments -dr and -d.
+  
+  Returns:
+    A list of filenames which match one of the extensions in
+    FILE_TYPE_EXTENSIONS.
+  """
+  filenames = []
+  if FLAGS.dr:
+    if FLAGS.verbose:
+      print()
+      print("Adding files by recursivly traversing directories:")
+    for directory in FLAGS.dr:
+      if FLAGS.verbose:
+        print("\tRecursive searching directory: \""+directory+"\"")
+      filenames += list_files_from_directory(directory, True)
+  if FLAGS.d:
+    for directory in FLAGS.d:
+      print("\Searching files in directory: \""+directory+"\"")
```

Comment:
Awesome feature of Python, you can use single, double, triple quotes for strings. And if any other quotes are within the string, they dont have to be escaped.

```
s = 'Recursively searching directory: "{0}"'.format(directory)
```